primitives/linux-port/column.cpp

```diff
@@ -1284,6 +1295,58 @@ inline SIMD_WRAPPER_TYPE simdSwapedOrderDataLoad(const ColRequestHeaderDataType
     return {result};
 }
 
+template <typename VT, typename SimdType>
+void vectorizedUpdateMinMax(const bool validMinMax, const MT nonNullOrEmptyMask, VT simdProcessor,
+                            SimdType& dataVec, SimdType& simdMin, SimdType& simdMax)
+{
+  if (validMinMax)
+  {
+    simdMin = simdProcessor.blend(
+        simdMin, dataVec,
+        simdProcessor.bwAnd(simdProcessor.cmpGt2(simdMin, dataVec),
+                            bitCast<SimdType>(simd::bitMaskToByteMask16(nonNullOrEmptyMask))));
+    simdMax = simdProcessor.blend(
+        simdMax, dataVec,
+        simdProcessor.bwAnd(simdProcessor.cmpGt2(dataVec, simdMax),
+                            bitCast<SimdType>(simd::bitMaskToByteMask16(nonNullOrEmptyMask))));
+  }
+}
+
+template <typename VT, typename SimdType>
+void vectorizedTextUpdateMinMax(const bool validMinMax, const MT nonNullOrEmptyMask, VT simdProcessor,
+                                SimdType& dataVec, SimdType& simdMin, SimdType& simdMax,
+                                SimdType& swapedOrderDataVec, SimdType& weightsMin, SimdType& weightsMax)
+{
+  if (validMinMax)
+  {
+    auto minComp = simdProcessor.bwAnd(simdProcessor.cmpGt2(weightsMin, swapedOrderDataVec),
+                                       bitCast<SimdType>(simd::bitMaskToByteMask16(nonNullOrEmptyMask)));
+    auto maxComp = simdProcessor.bwAnd(simdProcessor.cmpGt2(swapedOrderDataVec, weightsMax),
+                                       bitCast<SimdType>(simd::bitMaskToByteMask16(nonNullOrEmptyMask)));
+    simdMin = simdProcessor.blend(simdMin, dataVec, minComp);
+    weightsMin = simdProcessor.blend(weightsMin, swapedOrderDataVec, minComp);
+    simdMax = simdProcessor.blend(simdMax, dataVec, maxComp);
+    weightsMax = simdProcessor.blend(weightsMax, swapedOrderDataVec, maxComp);
+  }
+}
+
+template<typename T, typename VT, typename SimdType>
+void extractMinMax(VT& simdProcessor, SimdType simdMin, SimdType simdMax, T& min, T& max)
+{
+  constexpr const uint16_t size = VT::vecByteSize / sizeof(T);
+  T* simdMinVec = reinterpret_cast<T*>(&simdMin);
+  T* simdMaxVec = reinterpret_cast<T*>(&simdMax);
+  max = *std::max_element(simdMaxVec, simdMaxVec + size);
+  min = *std::min_element(simdMinVec, simdMinVec + size);
+}
+
+template <typename T, typename VT, typename SimdType>
+void getInitialSimdMinMax(VT& simdProcessor, SimdType& simdMin, SimdType& simdMax, T min, T max)
```

Comment:
You are the functional guy not me and I would prefer to return a pair from this getInitialSimdMinMax and break it into to separate vars using the structured binding feature.

Reply:
Actually since all this function does is a simple assignment, maybe I'll just do it right in the vectorizedFiltering? This will also be more in line with other var declarations.